src/Control/Monad/Free/Trans.purs

```diff
@@ -119,6 +120,16 @@ bimapFreeT :: forall f g m n a. Functor f => Functor n => (f ~> g) -> (m ~> n) -
 bimapFreeT nf nm (Bind e) = runExists (\(Bound a f) -> bound (bimapFreeT nf nm <<< a) (bimapFreeT nf nm <<< f)) e
 bimapFreeT nf nm (FreeT m) = FreeT \_ -> map (nf <<< map (bimapFreeT nf nm)) <$> nm (m unit)
 
+
+-- | Like `runFreeT`, but for running into some other FreeT without the
+-- | overhead that `MonadRec` incurs.
+substFreeT :: forall a m f g. Monad m => Functor g => (f ~> FreeT g m) -> FreeT f m a -> FreeT g m a
```

Comment:
NOTE: `Monad m` is needed because of `pure val`

Reply:
Not `Applicative`?

Reply:
`Monad m => Applicative (FreeT f m)` because Apply is implemented using `ap`